server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamLifecycle.java

```diff
@@ -131,15 +133,52 @@ public boolean isEnabled() {
     /**
      * The least amount of time data should be kept by elasticsearch.
      * @return the time period or null, null represents that data should never be deleted.
+     * @deprecated use {@link #getEffectiveDataRetention(DataStreamGlobalRetention)}
      */
+    @Deprecated
```

Comment:
Any reason to not just delete this method? It's not part of any public API, and it looks like it's only used in 2 tests.

Reply:
It will be easier to track down the places that it is being used to properly update them in the follow up PR, because I will be able to use the `Find usages`.

It's a convenience thing and they will be removed in the next PR.

Reply:
Future PR sounds good.